### PR TITLE
[FRONTEND] Messaging system

### LINKED ISSUE #356: [FRONTEND] Implement messages
- Implement messages between customer and vendors
- Will consist of conversation list and conversation pane
- Polling will be done (no websockets)

### PR DESCRIPTION
**Pull request by MehdiSaffar**
## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Description

- Add messages tab to profile
- Pane to show conversation list
- Pane to show current conversation messages
- Ability to send a regular text message or a message with single image as attachment (sent in base64)
- Add mock handlers for both getting all messages and sending a new message
- Loading shows a skeleton instead of spinner
- Polls every 5 seconds
- Shows last updated at

# Screenshots
image

image

# Test results

Manual tests

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have checked my code and corrected any misspellings
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# Potential problems/Important remarks

- Needs polling? (polls every 5 seconds)
- Needs refresh button?
- Need to show current user (done)
- How to handle base64 properly?
- How to always scroll down message list? (apparently react-chat-elements handles that with toBottomHeight prop)

closes #356

### CODE REVIEW

**Comment by MehdiSaffar** [comment]
add image preview

image

**Comment by MehdiSaffar** [comment]
Show empty when no conversation exists
image

Show empty when no conversation is selected
image

**Comment by abdullahyldz** [comment]
@use-it/interval
image-blobber
react-chat-elements
libraries are not saved to package.json

File: ecommerce/frontend/src/components/messages/Conversations.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        fetch()
+    }, [])
+
+    if (conversations !== null && !conversations.length) {
+        return (
+            <div className="conversations-empty">
+                <Empty description={'You have no conversations'} />
+            </div>
+        )
+    }
```
Comment (abdullahyldz) [inline ecommerce/frontend/src/components/messages/Conversations.js]: To enter into if condition and display 'you have no conversations'
Both `conversations is not null` and `conversations.length is 0` true

`if(!conversations or !conversations.length)` could be more error tolerant.

File: ecommerce/frontend/src/components/messages/Conversations.less [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+.conversations-empty {
+    display: grid;
+    overflow: auto;
+    width: 100%;
+    min-height: 500px;
+    place-content: center;
+}
+
+.conversations-list {
```
Comment (abdullahyldz) [inline ecommerce/frontend/src/components/messages/Conversations.less]: Layout is looking nice and clean. 
I think a gray vertical separator between the conversations list and messages could be nice. As the one in tabs pane on left.
image
Also, upload and send button at the bottom can have 1-2px gap

**Review by abdullahyldz — APPROVED** [review]

File: ecommerce/frontend/src/components/messages/Conversations.js (hunk 1 above)
Comment (MehdiSaffar) [reply ecommerce/frontend/src/components/messages/Conversations.js]: Sure, it is just that `conversations` normally can only take null or array

File: ecommerce/frontend/src/components/messages/Conversations.less (hunk 2 above)
Comment (MehdiSaffar) [reply ecommerce/frontend/src/components/messages/Conversations.less]: 👍🏼

**Comment by MehdiSaffar** [comment]
> @use-it/interval
> image-blobber
> react-chat-elements
> libraries are not saved to package.json

are you sure? I can see them
image

**Comment by MehdiSaffar** [comment]
- Add borders
- Better error handling
- Restrict to images
- Show preview of upload
- Allow delete of image

image

image

**Review by ozdenizdolu — APPROVED** [review]
Well written and I like your design a lot. Good job.

**Pull request merged**